vnext/Microsoft.ReactNative/Fabric/Composition/TextInput/WindowsTextInputComponentView.cpp

```diff
@@ -865,6 +884,11 @@ void WindowsTextInputComponentView::OnCharacterReceived(
     return;
   }
 
+  // save the last key pressed for onKeyPress
+  wchar_t key[2] = L" ";
+  key[0] = static_cast<wchar_t>(args.KeyCode());
+  m_lastKeyPressed = ::Microsoft::Common::Unicode::Utf16ToUtf8(key, 1);
```

Comment:
Any reason we dont just fire the onKeyPress event here?

Reply:
Hmm.. I just checked in expo snack.  On web and iOS onKeyPress fires before onChange.  On android onChange fires first.

So it doesn't look like the order is properly defined right now.

I think I'd go with the simpler code. Assuming that it is simpler to just fire the event in OnCharacterReceived/OnKeyDown

Reply:
If we don't care about matching Paper exactly for the order of events, I can switch it back! It does make the code _a lot_ more simpler. And actually reading the specs for TextInput, it looks like it should fire before (https://reactnative.dev/docs/textinput#onkeypress) so Paper may have a bug.

@jonthysell Would it be helpful to have a running document of intentional API changes between Paper/Fabric? I feel like it may make switching to Fabric easier but just an idea!